code/ui/manager/src/components/sidebar/Tree.tsx

```diff
@@ -231,40 +318,45 @@ const Node = React.memo<NodeProps>(function Node({
     );
   }
 
-  const BranchNode = item.type === 'component' ? ComponentNode : GroupNode;
-  return (
-    <BranchNode
-      key={id}
-      id={id}
-      className="sidebar-item"
-      data-ref-id={refId}
-      data-item-id={item.id}
-      data-parent-id={item.parent}
-      data-nodetype={item.type === 'component' ? 'component' : 'group'}
-      data-highlightable={isDisplayed}
-      aria-controls={item.children && item.children[0]}
-      aria-expanded={isExpanded}
-      depth={isOrphan ? item.depth : item.depth - 1}
-      isComponent={item.type === 'component'}
-      isExpandable={item.children && item.children.length > 0}
-      isExpanded={isExpanded}
-      onClick={(event) => {
-        event.preventDefault();
-        setExpanded({ ids: [item.id], value: !isExpanded });
-        if (item.type === 'component' && !isExpanded) onSelectStoryId(item.id);
-      }}
-      onMouseEnter={() => {
-        if (item.isComponent) {
-          api.emit(PRELOAD_ENTRIES, {
-            ids: [item.children[0]],
-            options: { target: refId },
-          });
-        }
-      }}
-    >
-      {(item.renderLabel as (i: typeof item) => React.ReactNode)?.(item) || item.name}
-    </BranchNode>
-  );
+  if (item.type === 'component' || item.type === 'group') {
```

Comment:
Isn't this check redundant? TS says so. I would prefer to throw if the runtime type doesn't match, don't you think?

Reply:
Yes it's redundant, I thought it made the code more clear.

Are you saying I should remove the if-statement, or add a `throw` at the end, if there were no matches?

Reply:
If you think it's clearer, I guess that's good. But yeah, I think throwing at the end rather than `null` is probably better if TS thinks its impossible. WDYT?

Reply:
It should be impossible, but I see no reason to risk throwing an error. Because throwing here would mean react completely unmounts the whole manager app.